packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -320,6 +334,19 @@ export class CallContext {
     });
   }
 
+  /* @conditional-compile-remove(rooms) */
+  public setParticipantRole(callId: string, participantKey: string, role: ParticipantRole): void {
```

Comment:
One minor thing to confirm, will participant events always happen before the RoleChange event? Just a bit worry about role event happening to soon so no participant could be found

Reply:
That's a really good thought. I will confirm that.